lib/player.js

```diff
@@ -5691,6 +5701,49 @@ shaka.Player = class extends shaka.util.FakeEventTarget {
     }
   }
 
+  /**
+   * Callback for liveSync
+   *
+   * @private
+   */
+  onTimeUpdate_() {
+    // If the live stream has reached its end, do not sync.
+    if (!this.isLive()) {
+      return;
+    }
+    const seekRange = this.seekRange();
+    if (!Number.isFinite(seekRange.end)) {
+      return;
+    }
+    const currentTime = this.video_.currentTime;
+    if (currentTime < seekRange.start) {
+      // Bad stream?
+      return;
+    }
+    const liveSyncMaxLatency = this.config_.streaming.liveSyncMaxLatency;
+    const liveSyncPlaybackRate = this.config_.streaming.liveSyncPlaybackRate;
+    const playbackRate = this.video_.playbackRate;
+    const latency = seekRange.end - this.video_.currentTime;
+    let offset = 0;
+    // HLS in SRC= doesn't update every time the seekrange, so we need calculate
+    // a safe offset
+    if (this.loadMode_ == shaka.Player.LoadMode.SRC_EQUALS) {
+      const buffered = this.video_.buffered;
+      if (buffered.length > 0) {
+        const bufferedEnd = buffered.end(buffered.length - 1);
+        offset = Math.max(liveSyncPlaybackRate, bufferedEnd - seekRange.end);
```

Comment:
According to MDN, the lowest frequency you can expect for the `timeupdate` event is [around 4 hz](https://developer.mozilla.org/en-US/docs/Web/API/HTMLMediaElement/timeupdate_event).
So taking an entire second's worth of fast-playback as an offset might be overkill, if the expectation is that this is being called at least 4 times a second. Even if you want to be safe and assume that system load is high, `liveSyncPlaybackRate / 2` might still work.

Reply:
In my tests this does not always work on SmartTV, I prefer to be more conservative in this case.